### PR TITLE
Milestone 02 : Processing data in models

### PR DESCRIPTION
**Pull request by FaranoshAmini**
## In this pull request, I have implemented the following :

- [x]  Create models
- [x]  Add associations between tables
- [x]  Insert data in tables
- [x]  Add data query methods
- [x]  Follow Linters
- [x]  Follow Ruby best practices
- [x]  Document work in a professional way

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (Ismailco) [inline app/models/user.rb]: - [x] Kindly make sure to add foreign_keys for the has_many relationship e.g `has_many :posts, foreign_key: 'author_id'`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @FaranoshAmini, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Nice use of [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)
✔️ Clean code
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [x]  Your project is not deployed yet, Kindly consider removing the `live demo` section altogether from your README

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @FaranoshAmini 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FaranoshAmini** [comment]
@tresorsawasawa, and @Ismailco, Thanks for your kind review and feedback.

**Pull request merged**